ci/ray_ci/test_tester.py

```diff
@@ -11,9 +12,23 @@
     _get_all_test_query,
     _get_test_targets,
     _get_flaky_test_targets,
+    _get_tag_matcher,
 )
 
 
+def test_get_tag_matcher() -> None:
+    assert re.match(
+        # simulate shell character escaping
+        bytes(_get_tag_matcher("tag"), "utf-8").decode("unicode_escape"),
```

Comment:
why is this using `unicode_escape`?

Reply:
It evaluates the escape character in a string, for example, the string 'a\\nb' will be evaluated to 'a\nb', which is what happen to a string in a shell command